docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/OrdinaryLeastSquaresAdvanced.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.SamplesUtils;
+
+namespace Samples.Dynamic.Trainers.Regression
+{
+    public static class OrdinaryLeastSquaresAdvanced
+    {
+        // This example requires installation of additional nuget package <a href="https://www.nuget.org/packages/Microsoft.ML.Mkl.Components/">Microsoft.ML.Mkl.Components</a>.
+        // In this examples we will use the housing price dataset. The goal is to predict median home value.
+        // For more details about this dataset, please see https://archive.ics.uci.edu/ml/machine-learning-databases/housing/
+        public static void Example()
+        {
+            // Downloading a regression dataset from github.com/dotnet/machinelearning
+            string dataFile = Microsoft.ML.SamplesUtils.DatasetUtils.DownloadHousingRegressionDataset();
+
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext(seed: 3);
+
+            // Creating a data loader, based on the format of the data
+            // The data is tab separated with all numeric columns.
+            // The first column being the label and rest are numeric features
+            // Here only seven numeric columns are used as features
+            var dataView = mlContext.Data.LoadFromTextFile(dataFile, new TextLoader.Options
+            {
+                Separators = new[] { '\t' },
+                HasHeader = true,
+                Columns = new[]
+               {
+                    new TextLoader.Column("Label", DataKind.Single, 0),
+                    new TextLoader.Column("Features", DataKind.Single, 1, 6)
+                }
+            });
+
+            //////////////////// Data Preview ////////////////////
+            // MedianHomeValue    CrimesPerCapita    PercentResidental    PercentNonRetail    CharlesRiver    NitricOxides    RoomsPerDwelling    PercentPre40s
+            // 24.00              0.00632            18.00                2.310               0               0.5380          6.5750              65.20
+            // 21.60              0.02731            00.00                7.070               0               0.4690          6.4210              78.90
+            // 34.70              0.02729            00.00                7.070               0               0.4690          7.1850              61.10
```

Comment:
and this!

Reply:
Humm, I don't quite understand this comment.

---
In reply to: [275458742](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275458742) [](ancestors = 275458742)

Reply:
In your code no one printing actual content of the dataview/file, which is not align with how we write samples right now (right now we have code which prints something to console and //Expected output comments).
No one prints this lines, right?
So can we remove them.
And in general can we get rid of calling Download file and run in-memory sample instead?

You spend good chunk of time in attempts to convince me what everything should be IN-MEMORY  NO EXCEPTION.
It's weird to see what you actually have references on SampleUtils.

Make sense?

---
In reply to: [275581463](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275581463) [](ancestors = 275581463,275458742)

Reply:
This file is not a API sample at all.

---
In reply to: [275582648](https://github.com/dotnet/machinelearning/pull/3319#discussion_r275582648) [](ancestors = 275582648,275581463,275458742)